contrib/guix/libexec/build.sh

```diff
@@ -255,6 +253,10 @@ mkdir -p "$DISTSRC"
     # shellcheck disable=SC2086
     env CFLAGS="${HOST_CFLAGS}" CXXFLAGS="${HOST_CXXFLAGS}" LDFLAGS="${HOST_LDFLAGS}" \
     cmake -S . -B build \
+          `# Required to ignore the CMAKE_PREFIX_PATH environment` \
+          `# variable set by Guix, which works for native search paths,` \
+          `# but not for cross-compiling.` \
+          -DCMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH=OFF \
```

Comment:
If this is needed, can it be set ealier / in depends and then passed down as part of the configuration? I assume anything built with CMake in depends could have this same problem? Setting this at this point, would make it seem like our depends configuration is leaky. The toolchain file/configuration it produces should not be looking outside of depends for anything?

Reply:
>  for example the <PackageName>_DIR variable. 

Discussed this offline. Usage of `<PackageName>_DIR variable.` is basically just reintroducing `ALLOW_HOST_PACKAGES`, which is not something we need to support (in depends).